packages/metadata-service/src/index.ts

```diff
@@ -67,24 +70,59 @@ server.get<{ Params: { keyToAssetKey: string } }>(
   }
 );
 
-server.get<{ Params: { eccCompact: string } }>("/:eccCompact", async (request, reply) => {
-  const { eccCompact } = request.params;
-  // const bufferCompact = Buffer.from(Address.fromB58(eccCompact).publicKey)
-  // const [storage] = await hotspotStorageKey(bufferCompact);
-  // const assetId = new PublicKey((await provider.connection.getAccountInfo(storage)).data.subarray(8, 8 + 32));
-  const digest = animalHash(eccCompact);
+server.get<{ Params: { eccCompact: string } }>(
+  "/:eccCompact",
+  async (request, reply) => {
+    const { eccCompact } = request.params;
+
+    // TODO: Remove this once we can update compressed nft metadata
+    if (eccCompact?.length === 22) {
+      try {
+        if (!program) {
+          program = await init(provider);
+        }
 
-  return {
-    name: digest,
-    description: "A Hotspot NFT on Helium",
-    image:
-      "https://shdw-drive.genesysgo.net/6tcnBSybPG7piEDShBcrVtYJDPSvGrDbVvXmXKpzBvWP/hotspot.png",
-    attributes: [
-      { trait_type: "ecc_compact", value: eccCompact },
-      { trait_type: "rewardable", value: true },
-    ],
-  };
-});
+        const [keyToAssetK] = keyToAssetKey(
+          daoKey(HNT_MINT)[0],
+          eccCompact,
+          "b58"
+        );
+
+        const keyToAsset = await program.account.keyToAssetV0.fetchNullable(
+          keyToAssetK
+        );
+
+        const asset = await getAsset(
```

Comment:
Why is any of this needed? Should just be a passthrough right?

Reply:
Ahh yeah good point